### ISSUE TITLE
macOS: Restore last session when closing window

### ISSUE THREAD

**Issue by danfy** [microsoft/vscode#173667]
Type: <b>Bug</b>

Virtual Studio Code in my Macbook does not reopen my last active session when reopening even with following settings: files.hotExit: "onExit" and window.restoreWindows: "all"

To clarify, VSCode does restore last active session after completely closing the app with command+Q, but not after pressing the red button. When I reopen the app after red button, the app only shows the blank page.

I tried both universal version and arm64 version (1.75 for both), but the problem still exists.

Steps to Reproduce:
1. Open several tabs to work on.
2. Close the app with red button (NOT WITH command + Q)
3. Reopen the app only to see the blank page

VS Code version: Code 1.75.0 (e2816fe719a4026ffa1ee0189dc89bdfdbafb164, 2023-02-01T15:26:31.005Z)
OS version: Darwin arm64 22.3.0
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Pro (12 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 4, 5|
|Memory (System)|16.00GB (2.07GB free)|
|Process Argv|--crash-reporter-id f2e753e7-40fd-4fe7-9563-c476303f0dc1|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
isort|ms-|2022.8.0
python|ms-|2023.2.0
vscode-pylance|ms-|2023.2.10
jupyter|ms-|2023.1.2000312134
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.14
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593:30376534
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
pyindex848:30577860
nodejswelcome1:30587005
2e4cg342:30602488
pyind779:30657576
f6dab269:30613381
pythonsymbol12cf:30657549
a9j8j154:30646983

```

</details>

**Comment by bpasero** [comment]
This works as designed currently, so it is not a bug. Treating as feature request.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by danfy** [comment]
I don't see how it is an intended feature in two reason. 1) I have tested this is several other computers and none of those computers did something like this; vscode in every single one of them restored the last active session when reopened, regardless of OS and regardless of whether I pressed a red button or completely closed the app with command+Q. 2) Even if it was an intended feature, who would implement such an inconsistent feature that would show the blank window when I reopen the app after the red button while showing the last active session after completely closing the app?

Now, if it was indeed an intended feature, how can I change the setting then?
